### PR TITLE
kerberos+refusing queries without msisdn filter

### PR DESCRIPTION
**Pull request by vbogdanovsky**

### CODE REVIEW

File: src/main/java/org/vroyer/hive/solr/SolrServerFactory.java [hunk 1]
```diff
@@ -21,12 +31,27 @@ private SolrServerFactory() {
     }
 
     public synchronized CloudSolrServer createCloudServer(String zkHost, String collectionId) {
+        CustomLogger.info("urlToCloudSolrServer.containsKey(" + collectionId + "): " + urlToCloudSolrServer.containsKey(collectionId));
         if (!urlToCloudSolrServer.containsKey(collectionId)) {
+            HttpClientUtil.setConfigurer(new Krb5HttpClientConfigurer());
```
Comment (vbogdanovsky) [inline src/main/java/org/vroyer/hive/solr/SolrServerFactory.java:36]: kerberos

**Comment by abbggg** [comment]
applied in #8

File: src/main/java/org/vroyer/hive/solr/SolrTableCursor.java [hunk 2]
```diff
@@ -110,13 +111,37 @@ private void fetchNextDocumentChunk() throws IOException {
...
+			CustomLogger.info("SolrTableCursor: PING to SOLR is OK: "+ response1);
+		}catch (SolrServerException e) {
+			CustomLogger.info("SolrTableCursor: ERROR solr PING: " + e);
+		} catch (RuntimeException e) {
+			CustomLogger.info("SolrTableCursor: RT ERROR solr PING: " + e);
+		}*/
+
+		CustomLogger.info("SolrTableCursor: Try to solr query: " + query);
+
+		String mandatoryFilterField = "msisdn_hex";
```
Comment (vbogdanovsky) [inline src/main/java/org/vroyer/hive/solr/SolrTableCursor.java:128]: START: Blocking queries without msisdn_hex

File: src/main/java/org/vroyer/hive/solr/SolrTableCursor.java [hunk 3]
```diff
@@ -110,13 +111,37 @@ private void fetchNextDocumentChunk() throws IOException {
...
+		if (!table.fq.toString().contains("msisdn_hex")) {
+			String error = "ERROR: Query to SOLR doesn't contain mandatory field [" + mandatoryFilterField + "]. The current SOLR filter (fq) is: [" + table.fq + "]. Stop the query.";
+			CustomLogger.info(error);
+			throw new IllegalArgumentException(error);
+		}
```
Comment (vbogdanovsky) [inline src/main/java/org/vroyer/hive/solr/SolrTableCursor.java:133]: END: Blocking queries without msisdn_hex

**Pull request closed without merging**